lib/rules/lint-no-action-helper.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const Rule = require('./base');
+
+const ERROR_MESSAGE =
+  'Do not use the `action` helper. Instead, use the `on` modifier and `fn` helper.';
+
+module.exports = class NoActionModifiers extends Rule {
+  visitor() {
+    let params = [];
+    function addParams(items) {
+      if (items.length) {
+        params = params.concat(items);
+      }
+    }
+    function removeParams(items) {
+      if (items.length) {
+        params = params.slice(0, -items.length);
+      }
+    }
+
+    function detectActionHelper(node) {
+      let maybeAction = node.path.original;
+      if (maybeAction !== 'action') {
+        return;
+      }
+      if (node.path.data === true || node.path.this === true) {
+        return;
+      }
+      if (params.includes(maybeAction)) {
```

Comment:
Should swap to using the built-in param tracking:

```suggestion
      if (this.isLocal(node)) {
```